docs/sdk/v3/guides/04-routing.md

```diff
@@ -3,4 +3,51 @@ id: routing
 title: Routing a Swap
 ---
 
-TODO
\ No newline at end of file
+## Introduction
+
+This guide will cover how to use the Uniswap Autorouter to compute optimal routes and execute swaps.
+It is based on the [Routing code example](https://github.com/Uniswap/examples/tree/main/v3-sdk/routing), found in the Uniswap code examples [repository](https://github.com/Uniswap/examples).
+To run this example, check out the guide's [README](https://github.com/Uniswap/examples/blob/main/v3-sdk/routing/README.md) and follow the setup instructions.
+
+:::info
+For a briefer on the SDK and to learn more about how these guides connect to the examples repository, please visit our [background](./01-background.md) page!
+:::
+
+In this example we will trade between **ETH and USDC**, but you can configure your example to us any two currencies and amount of input currency.
+
+The guide will **cover**:
+1. Getting a router instance
+2. Getting a route
+3. Swapping using a route
+
+At the end of the guide, we should be able to create a route and and execute a swap between any two currencies tokens using the example's included UI.
+
+## Getting a router instance
+
+To compute our route, we will use the `@uniswap/smart-order-router` package, specifically the `AlphaRouter` class which requires a chain ID and a provider. Note that routing is not supported for local forks, so we will use a mainnet provider even when swapping on a local fork.
```

Comment:
Should we be doing `:` before the examples?

Reply:
I think it makes sense in some cases but it doesn't after a context addition sentence like this one. I think it won't really trip anyone up as is but could confuse grammar if we try o uniformly apply